pulp_ansible/app/tasks/collections.py

```diff
@@ -448,6 +453,49 @@ async def _fetch_collection_metadata(self, requirements_entry):
 
         await asyncio.gather(*tasks)
 
+    async def _fetch_unpaginated_collection_metadata(self, name, namespace, requirement):
+        tasks = []
+        loop = asyncio.get_event_loop()
+
+        collection_metadata = filter(
+            lambda data: data["name"] == name and data["namespace"] == namespace,
+            self._unpaginated_collection_metadata,
```

Comment:
Algorithmically I think we should consider doing this differently. For a requirements file with N entries when there are M items to search through this results in O(N*M) performance.

Instead I think the downloading function should process the data before assigning it to `_unpaginated_collection_metadata` and `_unpaginated_collection_version_metadata`. It should organize the dictionaries with the top level keys being namespace and the second level key being collection_name.

Then these lookups can become just key/value lookups which would require just O(N + c) processing which is equivalent to O(N).